extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/write/HiveBatchWrite.scala

```diff
@@ -0,0 +1,273 @@
+              Some(ExternalCatalogUtils.generatePartitionPath(
+                partitionSpec,
+                partitionColumnNames,
+                tablePath))
+            } else {
+              oldPart.flatMap(_.storage.locationUri.map(uri => new Path(uri)))
+            }
+
+          import hive._
+          // SPARK-18107: Insert overwrite runs much slower than hive-client.
+          // Newer Hive largely improves insert overwrite performance. As Spark uses older Hive
+          // version and we may not want to catch up new Hive version every time. We delete the
+          // Hive partition first and then load data file into the Hive partition.
+          val hiveVersion = externalCatalog.asInstanceOf[ExternalCatalogWithListener]
+            .unwrapped.asInstanceOf[HiveExternalCatalog]
+            .client
+            .version
+          // SPARK-31684:
+          // For Hive 2.0.0 and onwards, as https://issues.apache.org/jira/browse/HIVE-11940
+          // has been fixed, and there is no performance issue anymore. We should leave the
+          // overwrite logic to hive to avoid failure in `FileSystem#checkPath` when the table
+          // and partition locations do not belong to the same `FileSystem`
+          // TODO(SPARK-31675): For Hive 2.2.0 and earlier, if the table and partition locations
+          // do not belong together, we will still get the same error thrown by hive encryption
+          // check. see https://issues.apache.org/jira/browse/HIVE-14380.
+          // So we still disable for Hive overwrite for Hive 1.x for better performance because
+          // the partition and table are on the same cluster in most cases.
+          if (partitionPath.nonEmpty && overwrite && hiveVersion < v2_0) {
+            partitionPath.foreach { path =>
+              val fs = path.getFileSystem(hadoopConf)
+              if (fs.exists(path)) {
+                if (!fs.delete(path, true)) {
+                  throw new RuntimeException(s"Cannot remove partition directory '$partitionPath'")
+                }
+                // Don't let Hive do overwrite operation since it is slower.
+                doHiveOverwrite = false
+              }
+            }
+          }
+
+          // inheritTableSpecs is set to true. It should be set to false for an IMPORT query
+          // which is currently considered as a Hive native command.
+          val inheritTableSpecs = true
+          val loadPath = ExternalCatalogUtils.generatePartitionPath(
+            staticPartitionSpec,
+            partitionColumnNames.take(partitionColumnNames.length - numDynamicPartitions),
+            tmpLocation.get)
+
+          externalCatalog.loadPartition(
+            table.database,
+            table.identifier.table,
+            loadPath.toString,
+            partitionSpec,
+            isOverwrite = doHiveOverwrite,
+            inheritTableSpecs = inheritTableSpecs,
+            isSrcLocal = false)
+        }
+      }
+    } else {
```

Comment:
move this branch to the head and just `return` instead of `if else` to make the code more readable